plutus-core/cost-model/data/models.R

```diff
@@ -650,6 +655,58 @@ modelFun <- function(path) {
     mkNilDataModel      <- constantModel ("MkNilData")
     mkNilPairDataModel  <- constantModel ("MkNilPairData")
 
+    ##### Show #####
+
+    showBoolModel <- {
+        fname <- "ShowBool"
+        filtered <- data %>%
+            filter.and.check.nonempty(fname) %>%
+            filter(x_mem == y_mem) %>%
+            discard.overhead ()
+        m <- lm(t ~ x_mem, filtered)
+        adjustModel(m,fname)
+    }
+
+    showIntegerModel <- {
+        fname <- "ShowInteger"
+        filtered <- data %>%
+            filter.and.check.nonempty(fname) %>%
+            filter(x_mem == y_mem) %>%
+            discard.overhead ()
+        m <- lm(t ~ x_mem, filtered)
+        adjustModel(m,fname)
+    }
+
+    showStringModel <- {
+        fname <- "ShowString"
+        filtered <- data %>%
+            filter.and.check.nonempty(fname) %>%
+            filter(x_mem == y_mem) %>%
+            discard.overhead ()
+        m <- lm(t ~ x_mem, filtered)
+        adjustModel(m,fname)
+    }
+
+    showByteStringModel <- {
+        fname <- "ShowByteString"
+        filtered <- data %>%
+            filter.and.check.nonempty(fname) %>%
+            filter(x_mem == y_mem) %>%
+            discard.overhead ()
+        m <- lm(t ~ x_mem, filtered)
+        adjustModel(m,fname)
+    }
+
+    showDataModel <- {
```

Comment:
We might have to do something like happens for `serialiseData` and `equalsData` here.   Maybe not though: if it's just showing the CBOR bytestring it should be quite well behaved, but if it's showing us what's inside the object things could be a lot more complicated.  I'm not sure what `show` does for `data` these days.

Reply:
Yeah, `Show Data` is derived, which is pretty bad (especially because of the `ByteString` it may contain). We should probably modify it to show the base64-encoded CBOR bytestring.

I wonder if we should also modify `Show ByteString` to show the base-64-encoded bytestring which looks nicer.